pentaho/src/main/java/pt/webdetails/cpf/olap/AbstractOlapUtils.java

```diff
@@ -216,11 +212,53 @@ protected Connection getMdxConnection( String catalog ) {
   }
 
   private Connection getMdxConnection( String catalog, String jndi ) {
+    MondrianCatalog selectedCatalog = getMondrianCatalog( catalog );
+
+    if ( selectedCatalog == null ) {
+      logger.error( "Received catalog '" + cleanLogString( catalog ) + "' doesn't appear to be valid" );
+      return null;
+    }
+
+    if ( !isJNDIValid( jndi ) ) {
+      logger.error( "Received jndi '" + cleanLogString( jndi ) + "' doesn't appear to be valid" );
+      return null;
+    }
+
     String connectStr = "provider=mondrian;dataSource=" + jndi + ";Catalog=" + catalog;
 
     return getMdxConnectionFromConnectionString( connectStr );
   }
 
+  private MondrianCatalog getMondrianCatalog( String catalog ) {
+    if ( catalog != null && catalog.startsWith( "/" ) ) {
+      catalog = StringUtils.substring( catalog, 1 );
+    }
+
+    return mondrianCatalogService.getCatalog( catalog, userSession );
+  }
+
+  private boolean isJNDIValid( String jndi ) {
+    boolean result = false;
+
+    if ( jndi != null ) {
+      try {
+        DataSource dataSourceImpl = getDatasourceImpl( jndi );
+
+        if ( dataSourceImpl != null ) {
+          result = true;
```

Comment:
why not just return instead of using a local variable?

Reply:
I did this to create a unique return point, but I can modify it or pass it to the parent method.